adapter/integration-test/src/adapter-datasource-trigger.test.js

```diff
@@ -0,0 +1,263 @@
+/* eslint-env jest */
+// @ts-check
+const request = require('supertest')
+const waitOn = require('wait-on')
+const amqp = require('amqplib')
+
+const URL = process.env.ADAPTER_API || 'http://localhost:9000/api/adapter'
+
+const RABBIT_HEALTH = `http://${process.env.RABBIT_HOST}:15672`
+const AMQP_URL = process.env.AMQP_URL
+const AMQP_EXCHANGE = process.env.AMQP_EXCHANGE
+const AMQP_IT_QUEUE = process.env.AMQP_IT_QUEUE
+const MOCK_SERVER_PORT = process.env.MOCK_SERVER_PORT || 8081
+const MOCK_SERVER_HOST = process.env.MOCK_SERVER_HOST || 'localhost'
+const MOCK_SERVER_URL = 'http://' + MOCK_SERVER_HOST + ':' + MOCK_SERVER_PORT
+const EXECUTION_TOPIC = process.env.AMQP_IMPORT_TOPIC
+const EXECUTION_SUCCESS_TOPIC = process.env.AMQP_IMPORT_SUCCESS_TOPIC
+const EXECUTION_FAILED_TOPIC = process.env.AMQP_IMPORT_FAILED_TOPIC
+
+let amqpConnection
+const publishedEvents = new Map() // routing key -> received msgs []
+
+describe('Adapter Sources Trigger', () => {
+  beforeAll(async () => {
+    console.log('Starting adapter sources trigger test')
+    const pingUrl = URL + '/version'
+    await waitOn({ resources: [MOCK_SERVER_URL, pingUrl, RABBIT_HEALTH], timeout: 50000, log: true })
+    console.log(`Services available. Connecting to amqp at ${AMQP_URL} ...`)
+    await connectAmqp(AMQP_URL)
+
+    await receiveAmqp(AMQP_URL, AMQP_EXCHANGE, EXECUTION_TOPIC, AMQP_IT_QUEUE)
+    console.log('Amqp connection established')
+  }, 60000)
+
+  afterAll(async () => {
+    if (amqpConnection) {
+      console.log('Closing AMQP Connection...')
+      await amqpConnection.close()
+      console.log('AMQP Connection closed')
+    }
+
+    // clear stored configs
+    await request(URL)
+      .delete('/configs')
+      .send()
+  })
+
+  test('POST datasources/{id}/trigger dynamic', async () => {
+    const datasourceResponse = await request(URL)
+      .post('/datasources')
+      .send(dynamicDatasourceConfig)
+    const datasourceId = datasourceResponse.body.id
+
+    const dataMetaData = await request(URL)
+      .post(`/datasources/${datasourceId}/trigger`)
+      .send(runtimeParameters)
+
+    const id = dataMetaData.body.id
+    const data = await request(URL)
+      .get(`/data/${id}`)
+      .send()
+
+    const delResponse = await request(URL)
+      .delete('/datasources/')
```

Comment:
Please only delete the created one, otherwise we won't come back to running the integration tests without `--runnInBands` flag at all

Reply:
yeah, good point